arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/NonEmptyList.kt

```diff
@@ -427,7 +425,8 @@ public inline fun <E, A, B> NonEmptyList<A>.traverse(f: (A) -> Either<E, B>): Ei
     }
   }
   // Safe due to traverse laws
-  return NonEmptyList.fromListUnsafe(acc).right()
+  return (acc.toNonEmptyListOrNull()
+    ?: throw IndexOutOfBoundsException("Empty list doesn't contain element at index 0.")).right()
```

Comment:
we should mention in the Docs that this throws an IndexOutOfBounds

Reply:
It was already doing that before with the unsafe access, I could've put `toNonEmptyListOrNull()!!` in the `ReplaceWith` but that would've broken the original signature in case some1 is relying on `try/catch` with `IndexOutOfBounds`.

This construction is actually safe here, so we might as well just use `!!`. We the resulting list from `traverse` will be `Nel`.